legend-engine-testable/src/main/java/org/finos/legend/engine/testable/TestableRunner.java

```diff
@@ -74,29 +73,32 @@ public RunTestsResult doTests(RunTestsInput input, MutableList<CommonProfile> pr
                 {
                     runTestsResult.results.add(testRunner.executeAtomicTest((Root_meta_pure_test_AtomicTest) test, pureModel, data));
                 }
-                if ((test instanceof Root_meta_pure_test_TestSuite) && (testIds.isEmpty() || testIdsBySuiteId.get(test._id()) != null))
+                if (test instanceof Root_meta_pure_test_TestSuite)
                 {
-                    Root_meta_pure_test_TestSuite testSuite = (Root_meta_pure_test_TestSuite) test;
-                    List<AtomicTestId> updatedTestIds;
-                    if (testIds.isEmpty())
+                    Map<String, List<AtomicTestId>> testIdsBySuiteId = testIds.stream().collect(groupingBy(testId -> testId.testSuiteId));
```

Comment:
+1 for reducing the scope of the variable

Reply:
Refactored this a bit. Running tests.